src/expand.cpp

```diff
@@ -205,7 +210,16 @@ static size_t parse_slice(const wchar_t *in, wchar_t **end_ptr, std::vector<long
             long tmp1 = fish_wcstol(&in[pos], &end);
             // Ignore the case of errno==-1 because it means the end char wasn't the null char.
             if (errno > 0) {
-                return pos;
+                // Check if we are at the last index expression, a missing end index means the range
```

Comment:
I'd like to see this check hoisted as well. If the next char is a ']', then the index is omitted; otherwise call wcstol()